pennylane/ops/op_math/adjoint.py

```diff
@@ -243,10 +243,6 @@ def __init__(self, base=None, do_queue=True, id=None):
     def __repr__(self):
         return f"Adjoint({self.base})"
 
-    # pylint: disable=protected-access
-    def _check_batching(self, params):
-        self.base._check_batching(params)
-
```

Comment:
We don't need this method.

Reply:
I think there was an argument that this keeps the batch_size attribute up to date. @albi3ro am I misremembering something here?

Reply:
The way I see it, any `SymbolicOp` wraps a PL operator that is already initialised, thus `self.base._check_batching(params)` is already called.

Maybe we need this when updating parameters dynamically...?

Reply:
Yes, exactly. So when updating the parameters of `Adjoint`, they might be broadcasted, but relying on `base.batch_size` later on will then report the batching status of the original parameters of the base, right?
The fact that you could remove it without noticing a problem means we're not testing for this, I suppose? :grimacing:

Reply:
Yes, we're definitely not testing it. But I don't know if we want to support updating the parameters, given that we are pushing towards having unmutable classes...

Reply:
Mhm. fair enough. Would it be a major pain point to leave this in, though? Just to not introduce a super nichy but possibly breaking change?

Reply:
Yep, maybe it's better to leave this in

Reply:
Added back!